src/fides/api/ops/util/connection_type.py

```diff
@@ -70,14 +73,65 @@ def connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]:
 
 
 def get_connection_types(
-    search: str | None = None, system_type: SystemType | None = None
+    search: str | None = None,
+    system_type: SystemType | None = None,
+    action_types: Set[ActionType] = SUPPORTED_ACTION_TYPES,
 ) -> list[ConnectionSystemTypeMap]:
     def is_match(elem: str) -> bool:
         """If a search query param was included, is it a substring of an available connector type?"""
         return search.lower() in elem.lower() if search else True
 
+    def saas_request_type_filter(connection_type: str) -> bool:
+        """
+        If any of the request type filters are set to true,
+        ensure the given saas connector supports requests of at least one of those types.
+        """
+        if SUPPORTED_ACTION_TYPES == action_types:
+            # if none of our filters are enabled, pass quickly to avoid unnecessary overhead
+            return True
+
+        template = ConnectorRegistry.get_connector_template(connection_type)
+        if template is None:  # shouldn't happen, but we can be safe
+            return False
```

Comment:
i put this here to satisfy the type checker, so making a test case for it seems like too much. i suppose i could have instead ignored the type checker, not sure what's better.